src/software/util/design_patterns/generic_factory.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+// TODO should be removed later
+#include <functional>
+//#include "software/backend/backend.h" // Should make generic.h
+
+
+// A quality of life typedef to make things shorter and more readable
+template <class GenericType>
+        using GenericRegistry = std::unordered_map<std::string, std::function<std::unique_ptr<GenericType>()>>;
+
+/**
+ * The GenericFactory is an Abstract class that provides an interface for Backend and Play
+ * Factories to follow. This makes it easy to maintain a list of factories and get the
```

Comment:
In the interests of minimizing coupling, the Backend and Play factories should know about this class, but this class should not know about them (ie. we shouldn't mention specific usages of the `GenericFactory` in the `GenericFactory`)

Reply:
Changed comments to reflect.